stdlib/Distributed/src/cluster.jl

```diff
@@ -636,7 +654,12 @@ function create_worker(manager, wconfig)
         # require the value of config.connect_at which is set only upon connection completion
         for jw in PGRP.workers
             if (jw.id != 1) && (jw.id < w.id)
-                (jw.state === W_CREATED) && wait(jw.c_state)
+                # wait for wl to join
+                if jw.state === W_CREATED
+                    lock(jw.c_state) do
```

Comment:
I think the lock should be around the `jw.state === W_CREATED` as well?

Reply:
Yes, Jameson recent comment reminded me that I got it wrong here